pynicotine/gtkgui/dialogs/preferences.py

```diff
@@ -2947,6 +2942,9 @@ def update_settings(self, settings_closed=False):
         self.application.window.chatrooms.toggle_chat_buttons()
         self.application.window.privatechat.toggle_chat_buttons()
 
+        if not config.sections["ui"]["spellcheck"]:
+            self.application.spell_checker = None
```

Comment:
This doesn't seem like a good idea if we still keep a spell checker around in the existing ChatEntry widgets.

Reply:
Especially not in combination with the call to init_spell_checker() above. Initializing a spell checker uses a fair bit of resources (especially memory).

Reply:
The spell checker that is initialized within `application` does not seem to be used anywhere, the changes in this PR would only re-initialize it (which I agree would be a good idea to avoid doing anyway).

> Initializing a spell checker uses a fair bit of resources

Would it perhaps be better to re-use the `application.spell_checker` object and re-assign the `spell_buffer` / `spell_view` in `on_switch_chat`?

Reply:
We only need to check spelling in one entry widget at a time, and with only a single buffer being active, since only 1 widget can have the input focus. Therefore multiple buffers are not needed.